gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-debug/src/main/java/io/gravitee/gateway/reactive/debug/context/DebugExecutionContext.java

```diff
@@ -0,0 +1,303 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.gateway.reactive.debug.context;
+
+import io.gravitee.el.TemplateEngine;
+import io.gravitee.el.TemplateVariableProvider;
+import io.gravitee.gateway.api.Request;
+import io.gravitee.gateway.api.Response;
+import io.gravitee.gateway.api.buffer.Buffer;
+import io.gravitee.gateway.api.http.HttpHeaders;
+import io.gravitee.gateway.core.component.ComponentProvider;
+import io.gravitee.gateway.debug.core.invoker.InvokerResponse;
+import io.gravitee.gateway.debug.reactor.handler.context.AttributeHelper;
+import io.gravitee.gateway.debug.reactor.handler.context.steps.DebugStep;
+import io.gravitee.gateway.policy.StreamType;
+import io.gravitee.gateway.reactive.api.ExecutionFailure;
+import io.gravitee.gateway.reactive.api.context.TlsSession;
+import io.gravitee.gateway.reactive.api.message.Message;
+import io.gravitee.gateway.reactive.api.tracing.Tracer;
+import io.gravitee.gateway.reactive.core.context.HttpExecutionContextInternal;
+import io.gravitee.gateway.reactive.core.context.MutableExecutionContext;
+import io.gravitee.gateway.reactive.core.context.MutableRequest;
+import io.gravitee.gateway.reactive.core.context.MutableResponse;
+import io.gravitee.gateway.reactive.debug.context.steps.DebugRequestStep;
+import io.gravitee.gateway.reactive.debug.context.steps.DebugResponseStep;
+import io.gravitee.reporter.api.v4.metric.Metrics;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.Maybe;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class DebugExecutionContext implements MutableExecutionContext {
+
+    private final MutableExecutionContext delegate;
+
+    private final LinkedList<DebugStep<?>> steps = new LinkedList<>();
+    private final InvokerResponse invokerResponse = new InvokerResponse();
+    private final Map<String, Serializable> initialAttributes;
+    private final HttpHeaders initialHeaders;
+
+    public DebugExecutionContext(MutableExecutionContext delegate) {
+        this.delegate = delegate;
+        this.initialAttributes = AttributeHelper.filterAndSerializeAttributes(delegate.getAttributes());
+        this.initialHeaders = HttpHeaders.create(request().headers());
+    }
+
+    public void beforePolicyExecution(DebugStep<?> debugStep) {
+        if (!steps.contains(debugStep)) {
+            steps.add(debugStep);
+
+            if (StreamType.ON_REQUEST.equals(debugStep.getStreamType())) {
+                ((DebugRequestStep) debugStep).before(request(), delegate.getAttributes());
+            } else {
+                ((DebugResponseStep) debugStep).before(response(), delegate.getAttributes());
+            }
```

Comment:
What do you think about:

```suggestion
            switch(debugStep) {
                case DebugRequestStep onRequest -> onRequest.before(request(), delegate.getAttributes());
                case DebugResponseStep onResponse -> onResponse.before(response(), delegate.getAttributes());
            }
```

Reply:
I've removed this class as it was useless